### PR TITLE
[WIP]Use latest version in ProgramLifecycleHandler

### PR DESCRIPTION
**Pull request by de-lan**

### CODE REVIEW

File: cdap-app-fabric/src/main/java/io/cdap/cdap/gateway/handlers/PreferencesHttpHandler.java [hunk 1]
```diff
@@ -170,8 +170,8 @@ public void deleteNamespacePrefs(HttpRequest request, HttpResponder responder,
   @GET
   public void getAppPrefs(HttpRequest request, HttpResponder responder,
                           @PathParam("namespace-id") String namespace, @PathParam("application-id") String appId,
-                          @QueryParam("resolved") boolean resolved) throws Exception {
```
Comment (jingtaowu3) [inline cdap-app-fabric/src/main/java/io/cdap/cdap/gateway/handlers/PreferencesHttpHandler.java:173]: should we keep the exception?

Comment (de-lan) [reply cdap-app-fabric/src/main/java/io/cdap/cdap/gateway/handlers/PreferencesHttpHandler.java:173]: But my IDE says it's no throw

Comment (jingtaowu3) [reply cdap-app-fabric/src/main/java/io/cdap/cdap/gateway/handlers/PreferencesHttpHandler.java:173]: interesting, so what happens if appId does not exist in db?

Comment (de-lan) [reply cdap-app-fabric/src/main/java/io/cdap/cdap/gateway/handlers/PreferencesHttpHandler.java:173]: I'll add it back, I saw many endpoints having this still

File: cdap-app-fabric/src/main/java/io/cdap/cdap/gateway/handlers/ProgramLifecycleHttpHandler.java [hunk 2]
```diff
@@ -1439,8 +1457,8 @@ public void getInstances(FullHttpRequest request, HttpResponder responder,
       // cant get instances for things that are not flows, services, or workers
       if (!canHaveInstances(runnable.getProgramType())) {
         output.add(new BatchRunnableInstances(runnable, HttpResponseStatus.BAD_REQUEST.code(),
-                   String.format("Program type '%s' is not a valid program type to get instances",
-                     runnable.getProgramType().getPrettyName())));
+                                              String.format("Program type '%s' is not a valid program type to get instances",
```
Comment (jingtaowu3) [inline cdap-app-fabric/src/main/java/io/cdap/cdap/gateway/handlers/ProgramLifecycleHttpHandler.java]: this fails stylecheck on my machine due to exceeding max characters

Comment (de-lan) [reply cdap-app-fabric/src/main/java/io/cdap/cdap/gateway/handlers/ProgramLifecycleHttpHandler.java]: cool, updated

**Pull request closed without merging**